BeatSaberPlaylistsLib/PlaylistManager.cs

```diff
@@ -220,9 +259,57 @@ public bool DeletePlaylist(IPlaylist playlist)
                 File.Delete(path);
                 var playlistToDelete = LoadedPlaylists.First(p => p.Value.Equals(playlist));
                 LoadedPlaylists.TryRemove(playlistToDelete.Key, out playlist);
-                return true;
+                return;
             }
-            return false;
+            throw new FileNotFoundException("Playlist not found in current manager.");
+        }
+
+        /// <summary>
+        /// Creates a new child <see cref="PlaylistManager"/>.
+        /// </summary>
+        /// <param name="folderName"></param>
+        /// <returns></returns>
+        /// <exception cref="IOException"></exception>
+        public PlaylistManager CreateChildManager(string folderName)
+        {
+            string newDirectory = Path.Combine(PlaylistPath, folderName);
+            Directory.CreateDirectory(newDirectory);
+            PlaylistManager childManager = new PlaylistManager(newDirectory, this);
+            ChildManagers.Add(childManager);
+            return childManager;
+        }
+
+        /// <summary>
+        /// Deletes a child and its directories recursively <see cref="PlaylistManager"/>.
+        /// </summary>
+        /// <param name="managerToDelete"></param>
+        /// <returns></returns>
+        /// <exception cref="IOException"></exception>
+        public void DeleteChildManager(PlaylistManager managerToDelete)
+        {
+            if (ChildManagers.Contains(managerToDelete))
+            {
+                Directory.Delete(managerToDelete.PlaylistPath, true);
+                ChildManagers.Remove(managerToDelete);
+            }
+            else
+            {
+                throw new FileNotFoundException("Folder not found under current manager");
+            }
+        }
+
+        /// <summary>
+        /// Renames the current manager folder <see cref="PlaylistManager"/>.
+        /// </summary>
+        /// <param name="folderName"></param>
+        /// <returns></returns>
+        /// <exception cref="IOException"></exception>
+        public void RenameManager(string folderName)
+        {
+            folderName.Replace("\\", "").Replace("/", "");
```

Comment:
See what happens if you move to an existing folder. If it throws an exception it's fine, otherwise throw one yourself.